runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5676,6 +5687,23 @@ void remote_get_buff_flush(void) {
   rwlock_unlock(&get_buff_global_info.lock);
 }
 
+static
+inline
+void remote_get_buff_task_flush(void) {
+  if (chpl_task_canMigrateThreads()) {
+    remote_get_buff_flush();
+  } else {
+    get_buff_thread_info_t* info = &get_buff_thread_info;
+    // Safe to check inited/vi outside the lock because no other thread can be
+    // modifying them, but concurrent flushing from other threads is possible.
+    if (info->inited && info->vi > 0) {
+      spinlock_lock(&info->lock);
+      get_buff_thread_info_flush(info);
```

Comment:
Should we test `info->vi > 0` again after we've acquired the spinlock, at least if tasking layer tasks can switch threads, in case while we were acquiring the spinlock (and yielding) some other task on the same thread popped in and flushed this thread's info?

Reply:
`get_buff_thread_info_flush` checks `info->vi > 0`